core/src/main/java/org/eclipse/hono/cache/CacheProvider.java

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.cache;
+
+public interface CacheProvider {
+
+    public <K, V> ExpiringValueCache<K, V> getCache(String cacheName);
+}
```

Comment:
Can you please add proper JavaDoc? I also think that adding the `FunctionalInterface` annotation would fit nicely ...

Reply:
:+1: for the Javadoc. About the functional interface, I had that in the beginning, but removed it again because, I am not sure if this will stay the only method over time. We might additional methods over time. Or might not. But in the end there is currently only one implementor of this interface. So the benefit of being able to use lambdas would effect only one location.

Reply:
Ok, I see.